### PR TITLE
add service methods

### LINKED ISSUE #1052: CoreAPI service
Create Core API service with Document Service

### PR DESCRIPTION
**Pull request by vedhavyas**
Closes #1052

### CODE REVIEW

File: documents/service.go [hunk 1]
```diff
@@ -60,6 +60,12 @@ type Service interface {
 
 	// Update validates and updates the model and return the updated model
 	Update(ctx context.Context, model Model) (Model, jobs.JobID, chan bool, error)
+
+	// CreateMode creates a new model from the payload and initiates the anchor process.
```
Comment (mikiquantum) [inline documents/service.go]:
```suggestion
	// CreateModel creates a new model from the payload and initiates the anchor process.
```

File: documents/service.go [hunk 2]
```diff
@@ -60,6 +60,12 @@ type Service interface {
 
 	// Update validates and updates the model and return the updated model
```
Comment (mikiquantum) [inline documents/service.go]: This `Create` and `Update` will be removed as soon as we finalized the transition to coreapi svc, right?

File: documents/service.go [hunk 3]
```diff
@@ -340,3 +346,21 @@ func (s service) Update(ctx context.Context, model Model) (Model, jobs.JobID, ch
 func (s service) getService(model Model) (Service, error) {
 	return s.registry.LocateService(model.DocumentType())
 }
+
+func (s service) CreateModel(ctx context.Context, payload CreatePayload) (Model, jobs.JobID, error) {
+	srv, err := s.registry.LocateService(payload.Scheme)
```
Comment (mikiquantum) [inline documents/service.go]: nice 👍

File: httpapi/coreapi/service.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package coreapi
```
Comment (mikiquantum) [inline httpapi/coreapi/service.go]: This package will be all document agnostic, correct?

File: documents/service.go (hunk 2 above)
Comment (vedhavyas) [reply documents/service.go]: yeap

File: httpapi/coreapi/service.go (hunk 4 above)
Comment (vedhavyas) [reply httpapi/coreapi/service.go]: yeap

**Comment by vedhavyas** [comment]
@mikiquantum updated the ticket. Will handle job status while working on that specific API.

**Pull request merged**